analyzers/src/SonarAnalyzer.CSharp/Rules/MethodOverrideAddsParams.cs

```diff
@@ -34,22 +34,19 @@ public sealed class MethodOverrideAddsParams : SonarDiagnosticAnalyzer
         internal const string DiagnosticId = "S3600";
         private const string MessageFormat = "'params' should be removed from this override.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat);
+        private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
                     var method = (MethodDeclarationSyntax)c.Node;
                     var methodSymbol = c.SemanticModel.GetDeclaredSymbol(method);
 
-                    if (methodSymbol == null ||
-                        !methodSymbol.IsOverride ||
-                        methodSymbol.OverriddenMethod == null)
+                    if (methodSymbol is not { IsOverride: true }
```

Comment:
since you converted this to `is {...}` syntax, maybe we can refactor it a bit more like this:
```csharp
   if (methodSymbol is not { IsOverride: true } or { OverriddenMethod: null })
```

Not sure if it is going to hurt readability (`not` is applied only to the left operand of `or`)

Reply:
maybe I'm not that used yet to pattern matching but this is harder for me to understand